interface/wx/httprequest.h

```diff
@@ -0,0 +1,232 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        httprequest.h
```

Comment:
I wonder if we shouldn't call it `WebRequest` instead, as it's going to support (at least on some platforms) not just HTTP, but HTTPS and HTTP/2 too.

Reply:
In a class I based some of this on I used WebRequest but thought http could be more appropriate on this level. But on the other end `WebSockets `are also not called `HTTPSockets `. It would also help to differentiate from `wxHTTP`